### PR TITLE
Added jest reporter from boneyard 

### PR DESCRIPTION
**Pull request by sirishaNekkanti**
### Summary
Copy the jest reporter from boneyard and utilize that as a reporter
solves issue : https://github.com/cerner/terra-toolkit/issues/635

[CONTRIBUTORS.md]: ../blob/main/CONTRIBUTORS.md
[License]: ../blob/main/LICENSE

### CODE REVIEW

File: packages/jest-config-terra/package.json [hunk 1]
```diff
@@ -39,7 +39,10 @@
     "babel-jest": "^26.6.3",
     "identity-obj-proxy": "^3.0.0",
     "jest-environment-jsdom": "26.6.2",
-    "jest-mock": "^26.6.2"
+    "jest-mock": "^26.6.2",
+    "@cerner/terra-cli": "^1.7.0",
```
Comment (mjhenkes) [inline packages/jest-config-terra/package.json:43]: What do we uses terra-cli for here? Can we not use it?

Comment (sirishaNekkanti) [reply packages/jest-config-terra/package.json:43]: using it for `Logger`

**Review by ShettyAkarsh — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

File: packages/jest-config-terra/package.json (hunk 1 above)
Comment (benbcai) [reply packages/jest-config-terra/package.json:43]: Can you move @cerner/terra-cli and "@jest/reporters" to the top of the dependency list? Typically scoped dependencies with the @ are listed first alphabetically.

File: packages/jest-config-terra/src/reporters/verbose-reporter/TerraVerboseReporter.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
+const VerboseReporter = require('@jest/reporters/build/verbose_reporter').default;
+const stripAnsi = require('strip-ansi');
+const fs = require('fs');
+const path = require('path');
+const endOfLine = require('os').EOL;
+const { Logger } = require('@cerner/terra-cli');
+
+const LOG_CONTEXT = '[Terra-Toolkit:terra-verbose-reporter]';
```
Comment (benbcai) [inline packages/jest-config-terra/src/reporters/verbose-reporter/TerraVerboseReporter.js]: The logger context should be changed to use the prefix. See https://github.com/cerner/terra-toolkit/blob/main/packages/terra-functional-testing/src/services/wdio-selenium-docker-service.js#L8 for an example. The prefix should changed be `[jest-config-terra:TerraVerboseReporter]`.

File: packages/jest-config-terra/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 
 ## Unreleased
+## Added
+
+- Copied the jest reporter from boneyard and utilize that as a reporter
```
Comment (benbcai) [inline packages/jest-config-terra/CHANGELOG.md:6]:
```suggestion

* Added
  * Copied the jest reporter from boneyard and utilize that as a reporter
```

**Pull request merged**